cfme/tests/intelligence/chargeback/test_chargeback_long_term_rates.py

```diff
@@ -0,0 +1,475 @@
+    # and hosts is on.This would mess up our Chargeback calculations, so we are disabling C&U
+    # collection after data has been fetched for the last hour.
+
+    appliance.server.settings.disable_server_roles(
+        'ems_metrics_coordinator', 'ems_metrics_collector')
+
+    # Perfrom rollup of C&U data.
+    ret = appliance.ssh_client.run_rails_command(
+        "\"vm = Vm.where(:ems_id => {}).where(:name => {})[0];\
+        vm.perf_rollup_range(1.hour.ago.utc, Time.now.utc,'realtime')\"".
+        format(provider.id, repr(vm_name)))
+    assert ret.success, "Failed to rollup VM C&U data:".format(ret.out)
+
+    wait_for(verify_records_rollups_table, [appliance, provider], timeout=600,
+        message='Waiting for hourly rollups')
+
+    # Since we are collecting C&U data for > 1 hour, there will be multiple hourly records per VM
+    # in the metric_rollups DB table.The values from these hourly records are summed up.
+
+    with appliance.db.client.transaction:
+        result = (
+            appliance.db.client.session.query(rollups.id)
+            .join(ems, rollups.parent_ems_id == ems.id)
+            .filter(rollups.capture_interval_name == 'hourly', rollups.resource_name == vm_name,
+            ems.name == provider.name, rollups.timestamp >= date.today())
+        )
+
+    for record in appliance.db.client.session.query(rollups).filter(
+            rollups.id.in_(result.subquery())):
+        consumed_hours = consumed_hours + 1
+        average_storage_used = average_storage_used + record.derived_vm_used_disk_storage
+        if any([record.cpu_usagemhz_rate_average,
+           record.cpu_usage_rate_average,
+           record.derived_memory_used,
+           record.net_usage_rate_average,
+           record.disk_usage_rate_average]):
+            average_cpu_used_in_mhz = average_cpu_used_in_mhz + record.cpu_usagemhz_rate_average
+            average_memory_used_in_mb = average_memory_used_in_mb + record.derived_memory_used
+            average_network_io = average_network_io + record.net_usage_rate_average
+            average_disk_io = average_disk_io + record.disk_usage_rate_average
+
+    # Convert storage used in Bytes to GB
+    average_storage_used = average_storage_used * math.pow(2, -30)
+
+    return {"average_cpu_used_in_mhz": average_cpu_used_in_mhz,
+            "average_memory_used_in_mb": average_memory_used_in_mb,
+            "average_network_io": average_network_io,
+            "average_disk_io": average_disk_io,
+            "average_storage_used": average_storage_used,
+            "consumed_hours": consumed_hours}
+
+
+def resource_cost(appliance, metric_description, usage, description, rate_type,
+        consumed_hours, interval):
+    """Query the DB for Chargeback rates"""
+    tiers = appliance.db.client['chargeback_tiers']
+    details = appliance.db.client['chargeback_rate_details']
+    cb_rates = appliance.db.client['chargeback_rates']
```

Comment:
OPTIONAL: as far as I see some of chargeback functionality is available in api. So, it makes sense to check what is missing and ask devs to provide such functionality thru api.